File: 2021/1.py

data = list(map(int, open("1.txt").read().split("\n")))
count = 0

for i in range(1, len(data)):
  if data[i] > data[i-1]:
    count += 1

print("part1:", count)

prevWindow = sum(data[:3])
count = 0

for i in range(3, len(data)):
  if (prevWindow - data[i-3] + data[i]) > prevWindow:
    count += 1
  prevWindow = prevWindow - data[i-3] + data[i]


print("part2:", count)

File: 2022_Rust/1.rs
use std::fs;

fn main() {
    let file_path = "1.txt";
    let contents = fs::read_to_string(file_path).expect("Should have been able to read the file");

    let batches = contents.split("\n\n");
    let mut biggest_batch_size: i32 = 0;
    let mut second_biggest_batch_size: i32 = 0;
    let mut third_biggest_batch_size: i32 = 0;
    for batch in batches {
        let nums = batch.split("\n");
        let mut batch_sum: i32 = 0;
        for num in nums {
            batch_sum += num
                .parse::<i32>()
                .expect("Should have been able to convert to i32");
        }
        if batch_sum > biggest_batch_size {
            third_biggest_batch_size = second_biggest_batch_size;
            second_biggest_batch_size = biggest_batch_size;
            biggest_batch_size = batch_sum;
        } else if batch_sum > second_biggest_batch_size {
            third_biggest_batch_size = second_biggest_batch_size;
            second_biggest_batch_size = batch_sum;
        } else if batch_sum > third_biggest_batch_size {
            third_biggest_batch_size = batch_sum;
        }
    }
    println!("Part1: {}", biggest_batch_size);
    println!("Part2: {}", biggest_batch_size + second_biggest_batch_size + third_biggest_batch_size);
}
